internal/cmd/shell.go

```diff
@@ -82,15 +82,20 @@ coder sh front-end-dev cat ~/config.json`,
 
 func shell(cmd *cobra.Command, cmdArgs []string) error {
 	ctx := cmd.Context()
-	command := "sh"
-	args := []string{"-c"}
+	var command string
+	var args []string
 	if len(cmdArgs) > 1 {
+		command = "/bin/sh"
+		args = []string{"-c"}
 		args = append(args, strings.Join(cmdArgs[1:], " "))
 	} else {
 		// Bring user into shell if no command is specified.
 		shell := "$(getent passwd $(id -u) | cut -d: -f 7)"
-		name := "-$(basename " + shell + ")"
-		args = append(args, fmt.Sprintf("exec -a %q %q", name, shell))
+
+		// force bash for the '-l' flag to the exec built-in
+		command = "/bin/bash"
+		args = []string{"-c"}
+		args = append(args, fmt.Sprintf("exec -l %q", shell))
```

Comment:
I don't instinctively trust %q to handle shell quoting rules, but this seems like it's good.

Reply:
hahaha @cmoog was suspicious of this too, perhaps I should change it? the Terminal app in the frontend is not quoted, and it works fine